### PR TITLE
Sync v2-10-stable with v2-10-test(sync_v2_10_test) to release 2.10.3

### PR DESCRIPTION
**Pull request by utkarsharma2**
Time for 2.10.3rc1!

### CODE REVIEW

File: RELEASE_NOTES.rst [hunk 1]
```diff
@@ -21,6 +21,55 @@
 
 .. towncrier release notes start
 
+Airflow 2.10.3 (2024-10-23)
+---------------------------
+
+Significant Changes
+^^^^^^^^^^^^^^^^^^^
+
+- Add ``run_with_db_retries`` when the scheduler updates the deferred Task as failed to tolerate database deadlock issues. (#41429)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]: This should just be a bugfix. It's not noteworthy to end users really.

File: RELEASE_NOTES.rst [hunk 2]
```diff
@@ -21,6 +21,55 @@
...
+- Ensure DAG trigger form submits with updated parameters upon keyboard submit (#42487) (#42499)
+- Do not attempt to provide not ``stringified`` objects to UI via xcom if pickling is active (#42388) (#42486)
+- Fix the span link of task instance to point to the correct span in the scheduler_job_loop (#42430) (#42480)
+- Bugfix task execution from runner in Windows (#42426) (#42478)
+- Apply otel_service on metrics (#42242) (#42441)
+- Use ``selectinload`` in trigger (#40487) (#42351)
+
+Miscellaneous
+"""""""""""""
+- Temporarily limits WTForms to below 3.2.0 (#43233) (#43234)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]: We should wait for #43309 and remove this one.

**Review by jedcunningham — COMMENTED** [review]
The list could use a "does the title make sense for an end user" pass or 2, but overall looks good.

File: RELEASE_NOTES.rst (hunk 2 above)
Comment (potiuk) [reply RELEASE_NOTES.rst]: Agree - if we are to **just** remove the description - but cherry-pick all commits regardless. It's just easier to apply them all :D

Comment (kaxil) [reply RELEASE_NOTES.rst]: Backport PR: https://github.com/apache/airflow/pull/43318

Comment (kaxil) [reply RELEASE_NOTES.rst]: that PR is merged, so feel free to adjust the release notes now @utkarsharma2

File: RELEASE_NOTES.rst (hunk 1 above)
Comment (utkarsharma2) [reply RELEASE_NOTES.rst]: I too think so, but wasn't sure.

Comment (utkarsharma2) [reply RELEASE_NOTES.rst]: Moved it to bug-fix section, PTAL.

File: RELEASE_NOTES.rst (hunk 2 above)
Comment (utkarsharma2) [reply RELEASE_NOTES.rst]: Updated the release notes post cherry-picking the commits. PTAL.

**Comment by utkarsharma2** [comment]
> The list could use a "does the title make sense for an end user" pass or 2, but overall looks good.

@jedcunningham I made some changes to the PR title in changelogs, PTAL.

**Comment by kaxil** [comment]
@utkarsharma2 we need to get https://github.com/apache/airflow/pull/43348 in fyi

**Comment by utkarsharma2** [comment]
> @utkarsharma2 we need to get #43348 in fyi

Included it.

**Review by ephraimbuddy — APPROVED** [review]

**Review by phanikumv — APPROVED** [review]

**Comment by potiuk** [comment]
One comment. I think - at least before final realease we need to release all the providers or at least the "apache-airflow-providers-fab" - with the `Flask-Application-Builder == 4.5.2`. We already have a number of issues of people who either do not use constraints or use and update their own constraints 

* https://github.com/apache/airflow/issues/43228
* https://github.com/apache/airflow/issues/43270
* https://github.com/apache/airflow/issues/43356
* https://github.com/apache/airflow/issues/43375

And this is something we can only address by releasing the new FAB provider and making sure constraints are updated to point to it.

**Comment by potiuk** [comment]
cc: @eladkal

**Comment by utkarsharma2** [comment]
> One comment. I think - at least before final realease we need to release all the providers or at least the `apache-airflow-providers-fab` - with the `Flask-Application-Builder == 4.5.2`. We already have a number of issues of people who either do not use constraints or use and update their own constraints (because of WTForms release breaking FAB):
> 
>     * [WTForms 3.2.0 broke Airflow main #43228](https://github.com/apache/airflow/issues/43228)
> 
>     * [500 on any admin pages #43270](https://github.com/apache/airflow/issues/43270)
> 
>     * [WTForms==3.2.1 breaks UI #43356](https://github.com/apache/airflow/issues/43356)
> 
>     * [AttributeError: 'tuple' object has no attribute 'items' when entering UI Admin Variables Section #43375](https://github.com/apache/airflow/issues/43375)
> 
> 
> And this is something we can only address by releasing the new FAB provider and making sure constraints are updated to point to it.

Since we already have 1st wave of providers out, I'm proceeding with release.

**Comment by potiuk** [comment]
> Since we already have 1st wave of providers out, I'm proceeding with release.

Go ahead, we can always regenerate the constraints to include the new providers released - they are rather independent from the release and internal RC testing and we've already did it in the past.

**Pull request merged**